build.gradle

```diff
@@ -542,3 +542,9 @@ allprojects {
     }
   } 
 }
+
+gradle.projectsEvaluated {
+  task s3ThirdPartyTests {
+    dependsOn project.tasks.findByPath("plugins:repository-s3:check"), project.tasks.findByPath("x-pack:snapshot-tool:check")
```

Comment:
I think we can define `s3ThirdPartyTests` in the respective subproject and just invoke gradle with `s3ThirdPartyTests` which will automatically find all those tasks in subprojects.

Reply:
My fear was that with that approach it becomes a lot more magical what happens here. Currently, there is no separation between general tests and 3rd party tests in the snapshot-cli tool which should change once GCS logic is introduced. It seemed safer to run `check` here now and then adjust this later when we know the exact test targets we have in the tool?
Also, S3 3rd party tests on CI currently just invoke `check` as well ... so maybe change one thing at a time? :) 
@mark-vieira wdyt?

Reply:
👍 please do this instead. The `repository-s3` and `snapshot-tool` projects should define a `s3ThirdPartyTests` task instead.

Reply:
We follow this pattern many other places. Stuff like `preCommit` and `internalClusterTest` follow these conventions. It's much better than this awkward coupling and usage of `projectsEvaluated`.

Reply:
2 against 1, fixing it :)